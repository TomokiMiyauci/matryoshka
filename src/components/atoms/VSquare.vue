<template>
  <td @click="click" :class="{ placeable: isValidClass }" class="square">
    {{ value }}
  </td>
</template>

<script>
export default {
  props: {
    value: {
      type: String,
      default: null
    },

    isValidClass: {
      type: Boolean,
      default: false
    }
  },

  methods: {
    click() {
      if (!this.value) {
        this.$emit('click')
      }
    }
  }
}
</script>

<style scoped>
.square {
  width: 100px;
  height: 100px;
  max-width: 100px;
  max-height: 100px;
  font-size: 60px;
  border: 1px solid #ccc;
  text-align: center;
}

@keyframes active {
  0% {
    background: red;
  }
  100% {
    background: blue;
  }
}
.placeable {
  animation-name: active;
  animation-duration: 3s;
  animation-timing-function: linear;
  animation-iteration-count: infinite;
}
</style>
