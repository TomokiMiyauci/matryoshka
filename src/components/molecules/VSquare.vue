<template>
  <div class="square" @click="$emit('click', piece)">
    <slot></slot>
  </div>
</template>

<script>
import { createComponent } from '@vue/composition-api'

export default createComponent({
  props: {
    piece: {
      type: Object,
      require: true
    }
  }
})
</script>

<style scoped>
.square {
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  top: 0;
  margin: 1px;
  background-color: rgba(47, 150, 69, 0.795);
  border-radius: 5%;
  display: flex;
  align-items: center;
  justify-content: center;
}
</style>
