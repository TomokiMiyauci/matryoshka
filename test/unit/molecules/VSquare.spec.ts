import VueCompositionApi from '@vue/composition-api'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import VSquare from '~/components/molecules/VSquare.vue'

const localVue = createLocalVue()
localVue.use(VueCompositionApi)

describe('VSquare.vue', () => {
  it('should have props of piece', () => {
    const piece = { row: 0, col: 0, value: [] }
    const wrapper = shallowMount(VSquare, {
      localVue,
      propsData: {
        piece
      }
    })

    expect(wrapper.props().piece).toEqual(piece)
  })

  it('should recive object type through props of piece', () => {
    const piece = { row: 0, col: 0, value: [] }
    const wrapper = shallowMount(VSquare, {
      localVue,
      propsData: {
        piece
      }
    })

    expect(typeof wrapper.props().piece).toBe('object')
  })

  it('should have slot at inner child', () => {
    const wrapper = shallowMount(VSquare, {
      localVue,
      slots: {
        default: '<span>test</span>'
      }
    })
    const slotContent = wrapper.find('span')

    expect(slotContent.exists()).toBeTruthy()
    expect(slotContent.text()).toBe('test')
  })

  it('should trigger emit when clicked', () => {
    const wrapper = shallowMount(VSquare, {
      localVue
    })
    wrapper.trigger('click')

    expect(wrapper.emitted('click')).not.toBeUndefined()
  })

  it('should return value when emitted', () => {
    const piece = { row: 0, col: 0, value: [] }
    const wrapper = shallowMount(VSquare, {
      localVue,
      propsData: {
        piece
      }
    })
    wrapper.trigger('click')

    expect(wrapper.emitted('click')[0][0]).toEqual({ ...piece })
  })

  it('should renders the correct markup', () => {
    const wrapper = shallowMount(VSquare, {
      localVue
    })

    expect(wrapper).toMatchSnapshot()
  })
})
