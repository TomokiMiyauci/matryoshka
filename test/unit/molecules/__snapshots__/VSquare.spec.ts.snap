// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VSquare.vue should renders the correct markup 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
